lib/src/page_list_viewport_gestures.dart

```diff
@@ -357,18 +448,46 @@ class DeprecatedPanAndScaleVelocityTracker {
       PageListViewportLogs.pagesListGestures.fine(() =>
           " - this gesture started really fast. Assuming that this is a continuation. Previous pointer count: $_previousPointerCount. Current pointer count: ${details.pointerCount}");
       _isPossibleGestureContinuation = true;
+    } else if (_timeSinceLastGesture != null &&
+        _timeSinceLastGesture! <
+            Duration(milliseconds: ViewportThresholdsAndScales.timeBetweenRepeatedScrollGestures)) {
+      // If the gesture is not a continued gesture, analyze if it can
+      // be a repeated accelerated swipe.
+
+      // If the previous gesture was a swipe, which triggered a momentum simulation
+      // and it was in the y direction only, mark the gesture as potentially a repeated
+      // acceleration swipe.
+      if (_previosLaunchedWithMomentum &&
+          // Scaling gestures cannot trigger acceleration of the viewport
+          details.pointerCount == 1 &&
+          !(_launchVelocity.dx.abs() > 0) &&
+          _launchVelocity != Offset.zero) {
+        // The gesture is strong enough to potentially accelerate the viewport faster than usual.
+        _isPossibleRepeatedAcceleratedSwipe = true;
+        _previousLaunchVelocity = _launchVelocity;
+      }
     } else {
       PageListViewportLogs.pagesListGestures.fine(() => " - restarting velocity for new gesture");
       _isPossibleGestureContinuation = false;
-      _previousGesturePointerCount = details.pointerCount;
+      _isPossibleRepeatedAcceleratedSwipe = false;
       _launchVelocity = Offset.zero;
     }
 
     _previousPointerCount = details.pointerCount;
     _startFocalPosition = details.localFocalPoint;
+
+    // Reinitialize the last position tracker every new gesture
+    _lastPositions.clear();
+    _lastPositions.addFirst(_startFocalPosition);
   }
 
   void onScaleUpdate(Offset localFocalPoint, int pointerCount) {
+    // Update the queue tracking last positions
```

Comment:
I see the term "last" and "positions" used a number of places. But these are contradictory terms. "Last" is the singular, final or most recent thing. But "positions" is plural, implying there can be many.

Also, notice that your list refers to "positions" but what you're storing is "Offsets". Is there a reason that you chose a different term? If not, it's best to stay consistent.

I think I would name this `_focalPointHistory`.

You might also want to check if there's a special kind of list that automatically evicts when you push. I can't remember if Dart has that, but if it does, you can probably use that and avoid doing that yourself.

Reply:
A circular buffer would be optimal, but dart doesn't have that. Should I leave this implementation as is?